app/pb/communities.proto

```diff
@@ -0,0 +1,105 @@
+syntax = "proto3";
+
+package org.couchers.api.communities;
+
+import "google/protobuf/timestamp.proto";
+
+service Communities {
+  // This is a secure service: a user needs to be authenticated and not jailed to call functions here, refer to auth.proto and jail.proto
+
+  // Discussion threads are organized in three levels:
+  // thread1
+  // +- comment1
+  // |  +- reply1
+  // |  +- reply2
+  // +- comment2
+  //    +- reply3
+  //    +- reply4
+  // thread2
+  // +- comment3
+  // |  +- reply5
+  // |  +- reply6
+  // +- comment4
+  //    +- reply7
+  //    +- reply8
+
+  // Get non-deleted comments belonging to a thread
+  rpc GetComments(GetCommentsReq) returns (GetCommentsRes);
+
+  // Get non-deleted replies belonging to a comment
+  rpc GetReplies(GetRepliesReq) returns (GetRepliesRes);
+
+  // Post a new comment on a thread
+  rpc PostComment(PostCommentReq) returns (PostCommentRes);
+
+  // Post a new reply to a comment
+  rpc PostReply(PostReplyReq) returns (PostReplyRes);
+}
+
+message GetCommentsReq {
+  int64 thread_id = 1;
+  // Only get messages posted before the given comment id. Pass -1 to get all.
+  int64 posted_before_comment_id = 2;
```

Comment:
How about using 0 for unset? The IDs will come from postgres serials that start from 1, so IDs should never be 0?

Reply:
Additional benefit: protobuf ints are 0 by default